### PR TITLE
Matching effects

### PR DESCRIPTION
**Pull request by mrshigure**

### CODE REVIEW

File: src/effects/effect_3D.c [hunk 1]
```diff
@@ -1,14 +1,212 @@
...
+extern Gfx D_090002E8[];
+
+void fx_3D_init(EffectInstance* effect);
+void fx_3D_update(EffectInstance* effect);
+void fx_3D_render(EffectInstance* effect);
 void fx_3D_appendGfx(void* effect);
 
+// Some operations happening in the wrong order.
+// Offsets in the 'for' loop look wrong but they're just shifted by 8 bytes.
+#if NON_MATCHING
```
Comment (mrshigure) [inline src/effects/effect_3D.c:14]: Couldn't get this one right: https://decomp.me/scratch/NufxC

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 12 functions (+0.26%) on `us`.

File: src/effects/drop_leaves.c [hunk 2]
```diff
@@ -23,4 +126,33 @@ void drop_leaves_render(EffectInstance* effect) {
...
+void drop_leaves_appendGfx(void* effect) {
+    DropLeavesFXData* part = ((EffectInstance*)effect)->data.dropLeaves;
+    Matrix4f sp18;
+    Matrix4f sp58;
+    Matrix4f sp98;
+    s32 i;
+
+    gSPSegment(gMasterGfxPos++, 0x09, VIRTUAL_TO_PHYSICAL(((EffectInstance*)effect)->graphics->data));
+    gSPDisplayList(gMasterGfxPos++, D_09001180);
+    gDPSetPrimColor(gMasterGfxPos++, 0, 0, 0x14, 0x64, 0x14, part->unk_24);
```
Comment (ethteck) [inline src/effects/drop_leaves.c:138]: decimal colors pleez

**Review by ethteck — CHANGES_REQUESTED** [review]
Excellent shtuff

**Review by ethteck — APPROVED** [review]

**Comment by ethteck** [comment]
> Wondering if the `_appendGfx` funcs would be better off taking `EffectInstance*` rather than `void*`.. @ethteck?

We can do this later if it works out, but there's evidence they take void* due to needing a temp for a0 in many of these functions

**Review by nanaian — APPROVED** [review]

**Pull request merged**